actors/market/src/lib.rs

```diff
@@ -370,40 +356,59 @@ impl Actor {
                 continue;
             }
 
-            // check VerifiedClient allowed cap and deduct PieceSize from cap
-            // drop deals with a DealSize that cannot be fully covered by VerifiedClient's available DataCap
-            if deal.proposal.verified_deal {
-                if let Err(e) = rt.send(
-                    &VERIFIED_REGISTRY_ACTOR_ADDR,
-                    crate::ext::verifreg::USE_BYTES_METHOD as u64,
-                    RawBytes::serialize(UseBytesParams {
-                        address: Address::new_id(client_id),
-                        deal_size: BigInt::from(deal.proposal.piece_size.0),
-                    })?,
-                    TokenAmount::zero(),
-                ) {
-                    info!("invalid deal {}: failed to acquire datacap exitcode: {}", di, e);
-                    continue;
+            // For verified deals, transfer datacap tokens from the client
+            // to the verified registry actor along with a specification for the allocation.
+            // Drop deal if the transfer fails.
+            // This could be done in a batch, but one-at-a-time allows dropping of only
+            // some deals if the client's balance is insufficient, rather than dropping them all.
+            let allocation_id = if deal.proposal.verified_deal {
+                let params = datacap_transfer_request(
+                    &Address::new_id(client_id),
+                    vec![alloc_request_for_deal(&deal, rt.policy(), curr_epoch)],
+                )?;
+                let alloc_ids = rt
+                    .send(
+                        &DATACAP_TOKEN_ACTOR_ADDR,
+                        ext::datacap::TRANSFER_FROM_METHOD as u64,
+                        serialize(&params, "transfer parameters")?,
+                        TokenAmount::zero(),
+                    )
+                    .and_then(|ret| datacap_transfer_response(&ret));
+                match alloc_ids {
+                    Ok(ids) => {
+                        if ids.len() != 1 {
+                            return Err(actor_error!(
+                                illegal_argument,
```

Comment:
This is more like illegal return.  Is it still easy after move to fvm to define actor exit codes?  It might be worth it.

Reply:
This sort of error where the thing you call is broken started another conversation in recent review which is why it seems general enough to be worth it to me.

Reply:
Actor-specific exit codes? Yes sure, still easy and generally good. I'll change this to `USR_UNSPECIFIED`, but happy to change to an actor code in follow-up.